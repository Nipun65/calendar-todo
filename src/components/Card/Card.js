import styles from "./Card.module.css";
import { handleFunc } from "../Handlers/Handlers";

const Card = ({ data, setValue }) => {
  return (
    <div className={styles["viewgrid"]}>
      {data.map((dataValue) => (
        <div
          onClick={(event) => handleFunc(setValue, event.target.textContent)}
          className={styles["element"]}
          key={Math.random()}
        >
          {dataValue}
        </div>
      ))}
    </div>
  );
};
export default Card;
