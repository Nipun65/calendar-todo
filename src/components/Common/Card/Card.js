import styles from './Card.module.css';

function Card({ data, setValue }) {
  // to set selected element in parent component
  const valueHandler = (event) => {
    setValue(event.target.textContent);
  };

  return (
    <div className={styles.viewgrid}>
      {data.map((dataValue) => (
        <div
          role="presentation"
          onClick={valueHandler}
          className={styles.element}
          key={Math.random()}
          type="button"
        >
          {dataValue}
        </div>
      ))}
    </div>
  );
}

Card.defaultProps = { data: [] };
export default Card;
