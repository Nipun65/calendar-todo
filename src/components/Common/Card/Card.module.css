.viewgrid {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(4, 1fr);
  height: 85px;
}
.viewgrid :hover {
  background-color: rgb(60, 64, 67, 0.1);
}

.element {
  justify-content: center;
  height: 85px;
  padding-top: 2em;
}
