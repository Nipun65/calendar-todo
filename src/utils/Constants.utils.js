export const MONTHS = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December',
];

export const DROPDOWNLIST = ['Month', 'Year'];

export const DAYS = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];

export const DATE = {
  year: 2023,
  month: 0,
  date: 31,
  currentMonth: 0,
  currentYear: 2023,
  selectedDate: { userYear: 2023, userMonth: 0, userDate: 31 },
};
