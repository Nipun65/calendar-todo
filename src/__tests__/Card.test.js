import renderer from 'react-test-renderer';
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Card from '../components/Common/Card/Card';
import { MONTHS } from '../utils/Constants.utils';

afterEach(() => {
  cleanup();
});

it('should match the snapshot', () => {
  const tree = renderer.create(<Card data={MONTHS} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render Card Component', () => {
  render(<Card data={MONTHS} />);
  const mainDiv = screen.getByTestId('card');
  expect(mainDiv).toBeInTheDocument();
  const childCount = mainDiv.children.length;
  expect(childCount).toBe(12);
});
