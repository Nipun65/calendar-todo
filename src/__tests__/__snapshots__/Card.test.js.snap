// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="viewgrid"
  data-testid="card"
>
  <div
    className="element is-clickable is-flex"
    onClick={[Function]}
    role="presentation"
    type="button"
  >
    January
  </div>
  <div
    className="element is-clickable is-flex"
    onClick={[Function]}
    role="presentation"
    type="button"
  >
    February
  </div>
  <div
    className="element is-clickable is-flex"
    onClick={[Function]}
    role="presentation"
    type="button"
  >
    March
  </div>
  <div
    className="element is-clickable is-flex"
    onClick={[Function]}
    role="presentation"
    type="button"
  >
    April
  </div>
  <div
    className="element is-clickable is-flex"
    onClick={[Function]}
    role="presentation"
    type="button"
  >
    May
  </div>
  <div
    className="element is-clickable is-flex"
    onClick={[Function]}
    role="presentation"
    type="button"
  >
    June
  </div>
  <div
    className="element is-clickable is-flex"
    onClick={[Function]}
    role="presentation"
    type="button"
  >
    July
  </div>
  <div
    className="element is-clickable is-flex"
    onClick={[Function]}
    role="presentation"
    type="button"
  >
    August
  </div>
  <div
    className="element is-clickable is-flex"
    onClick={[Function]}
    role="presentation"
    type="button"
  >
    September
  </div>
  <div
    className="element is-clickable is-flex"
    onClick={[Function]}
    role="presentation"
    type="button"
  >
    October
  </div>
  <div
    className="element is-clickable is-flex"
    onClick={[Function]}
    role="presentation"
    type="button"
  >
    November
  </div>
  <div
    className="element is-clickable is-flex"
    onClick={[Function]}
    role="presentation"
    type="button"
  >
    December
  </div>
</div>
`;
